src/box/sql/build.c

```diff
@@ -799,9 +828,10 @@ sql_create_check_contraint(struct Parse *parser)
 			      space_id_reg);
 		vdbe_emit_ck_constraint_create(parser, ck_def, space_id_reg,
 					       space->def->name);
-		assert(sqlVdbeGetOp(v, v->nOp - 1)->opcode == OP_SInsert);
+		struct VdbeOp *op = sqlVdbeGetOp(v, v->nOp - 2);
+		assert(op->opcode == OP_SInsert);
 		sqlVdbeCountChanges(v);
-		sqlVdbeChangeP5(v, OPFLAG_NCHANGE);
+		op->p5 = OPFLAG_NCHANGE;
```

Comment:
Probably the same needs to be done in the patch about fkeys - count the successful ops as `OPFLAG_NCHANGE`. `row_count == 1` when success, otherwise if it wasn't an error then `== 0`. Regardless of the number of actually affected rows, AFAIR.

Reply:
Thanks, fixed. I dropped this code and added this functionality to opcode.